pkg/ebpf/c/tracee.bpf.c

```diff
@@ -656,6 +664,382 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
     return 0;
 }
 
+// If we won't create these maps, the userspace needs to be kernel version aware as well,
+// otherwise it'll try to interact with the maps and report an error
+BPF_HASH(unhidden_modules_map, u64, u64, 2048); // modules that are shown (not hidden)
+BPF_HASH(modules_from_procfs, u64, kernel_module_name_t, 2048); // modules returned by reading /proc
+
+void __always_inline send_hidden_module(u64 mod_addr, char *mod_name, program_data_t *p)
+{
+    reset_event_args(p);
+    save_to_submit_buf(p->event, &mod_addr, sizeof(u64), 0);
+    save_bytes_to_buf(p->event, (void *) mod_name, MODULE_NAME_LEN & MAX_MEM_DUMP_SIZE, 1);
+
+    events_perf_submit(p, HIDDEN_KERNEL_MODULE_SEEKER, 0);
+}
+
+// Populate the queue with the current modules in the list of modules.
+// We can't read it once and then hook on do_init_module and free_module since a hidden module will
+// remove itself from the list directly and we wouldn't know (hence from our perspective the module
+// will reside in the modules list, which could be false). So on every trigger, we go over the
+// modules list and populate the queue. This is a queue since we need to delete all elements so the
+// next run will be clean. Since this mechanism is suppose to be triggered every once in a while,
+// this should be ok.
+static __always_inline bool init_shown_modules_queue()
+{
+    char modules_sym[8] = "modules";
+    struct list_head *head = (struct list_head *) get_symbol_addr(modules_sym);
+    bool iterated_all_modules = false;
+    struct module *pos, *n;
+    u64 empty = 0;
+
+    pos = list_first_entry_ebpf(head, typeof(*pos), list);
+    n = list_next_entry_ebpf(pos, list);
+
+#pragma unroll
+    for (int i = 0; i < MAX_NUM_MODS; i++) {
+        if (&pos->list == head) {
+            iterated_all_modules = true;
+            break;
+        }
+
+        bpf_map_update_elem(&unhidden_modules_map, &pos, &empty, BPF_ANY);
+
+        pos = n;
+        n = list_next_entry_ebpf(n, list);
+    }
+
+    return !iterated_all_modules; // false is valid value
```

Comment:
i think it would be clearer to just return 0 instead of breaking, and return 1 at the end of func. your choice though, its not a big deal..